share/spice/thesaurus/thesaurus.js

```diff
@@ -2,83 +2,90 @@
+            });
 
-        var headers = {
-            "syn" : "Synonyms of ",
-            "ant" : "Antonyms of ",
-            "rel" : "Related to ",
-            "sim" : "Similar to "
-        };
-
-        // Check if the mode exists.
-        var how_many = 0;
-        for(var i in api_result) {
-            if(api_result.hasOwnProperty(i) && (shorthand[mode] in api_result[i])) {
-                how_many += 1;
+            if (wordTypeEntryTotal > largestWordTypeEntryTotal) {
+                largestWordTypeEntryTotal = wordTypeEntryTotal;
+                wordTypeWithMostEntries = wordType;
             }
-        }
-        if(how_many === 0) {
-            return Spice.failed('thesaurus');
+        });
+        
+        if (!wordTypeWithMostEntries) {
+            return Spice.failed("thesaurus");    
         }
 
-        api_result.mode = shorthand[mode];
-
-        // Create the plugin.
         Spice.add({
             id: 'thesaurus',
             name: 'Thesaurus',
-            data:  api_result,
-            normalize: function(item){
-                var res = {
-                    headerText: headers[item.mode] || 'Thesaurus: ',
-                    query: query,
-                    results: []
-                };
-
-                for(var parts_of_speech in item) {
-                    if(item.hasOwnProperty(parts_of_speech) && item[parts_of_speech][item.mode]) {
-                        res.results.push({
-                            heading : parts_of_speech.charAt(0).toUpperCase() + parts_of_speech.slice(1),
-                            words   : item[parts_of_speech][item.mode].splice(0, 10).join(", ")
-                        });
-                    }
+            data:  api_result[wordTypeWithMostEntries],
+            templates: {
+                group: 'list',
+                options:{
+                    content: 'record', 
+                    rowHighlight : true,
+                    moreAt: true
                 }
-
-                return res;
             },
+            
             meta: {
                 sourceName:  'Big Huge Thesaurus',
                 sourceUrl:  'http://words.bighugelabs.com/' + query,
             },
-            templates: {
-                group: 'text',
-                options: {
-                    content: Spice.thesaurus.content,
-                    title_content: Spice.thesaurus.title_content,
-                    moreAt: true
+            
+            normalize: function(item) {
+                var RELATIONSHIP_TYPE = {
+                    SYNONYMS : "syn",
+                    SIMILAR  : "sim",                    
+                    RELATED  : "rel",
+                    ANTONYMS : "ant"
+                };
+                
+                var result = {
+                    title: query,
+                    subtitle: wordTypeWithMostEntries,                    
+                    record_data : {}
```

Comment:
@sekhavati you can use an array if you'd like, that way the order is preserved.

You can also create a `record_keys` array and the order of that will be used instead.

More details here: https://github.com/duckduckgo/duckduckgo-template-helpers/blob/67f7bb936ad825cf88c66665d0201d07251d707e/template_helpers.js#L851